### PR TITLE
fix(components): [input-number] Decimal point issue resolution

### LINKED ISSUE #23563: [Component] [input-number] el-input-number中文输入法时bug
Bug Type: **`Component`**

## Environment

- Vue Version: `3.5.26`
- Element Plus Version: `2.13.1`
- Browser / OS: `143.0.7499.170（正式版本）（64 位）/windown10 专业版`
- Build Tool: `Vite`

### Related Component

- `el-input-number`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtaW5wdXQtbnVtYmVyIHYtbW9kZWw9XCJudW1cIiA6cHJlY2lzaW9uPVwiMlwiIDpzdGVwPVwiMC4xXCIgOm1heD1cIjEwXCIgLz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IG51bSA9IHJlZigxKVxuPC9zY3JpcHQ+XG4iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0AyLjEzLjAvZGlzdC9pbmRleC5jc3MnLCAnaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAMi4xMy4wL3RoZW1lLWNoYWxrL2RhcmsvY3NzLXZhcnMuY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGxTZXR0bGVkKHN0eWxlcylcbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0AyLjEzLjAvZGlzdC9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0AyLjEzLjAvXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiX28iOnsiZXBWZXJzaW9uIjoiMi4xMy4wIn19)

### Steps to reproduce

**当输入法为中文时**：
**问题1：**
1、先输入小数
2、删除小数点
3、此时无法再输入小数点，除非在最后在输入一位数后才能输入小数点
**问题2：**
当输入最后有小数点，例如：“11.”时，失去焦点后，没做处理。

### What is Expected?

1、在中文输入法时，删除小数点后再输入小数点可以输入小数点
2、当输入最后有小数点，例如：“11.”时，失去焦点后，没做处理。

### What is actually happening?

1、在中文输入法时，第一次可以输入小数点，删除小数点后就不能再输入小数点了，除非在最后在输入一位数后才能输入小数点
2、输入最后是小数点，做处理

### Additional comments

_(empty)_

Image

### PR DESCRIPTION
**Pull request by E66Crisp**
Please make sure these boxes are checked before submitting your PR, thank you!

fix:  #23563
relation  #23565

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

## Summary by CodeRabbit

* **Bug Fixes**
  * Input now normalizes full-width dots to periods, strips invalid characters, and preserves/filters exponent characters based on scientific mode.
  * Rendering changed to a plain-text-style input for more consistent typing, sanitization, and event behavior.

* **New Features**
  * Added reusable input-sanitization so number entry is normalized before processing.

### CODE REVIEW

File: packages/components/input-number/src/input-number.vue [hunk 1]
```diff
@@ -323,6 +322,7 @@ const setCurrentValue = (
   data.currentValue = newVal
 }
 const handleInput = (value: string) => {
+  value = value.replace(/。/g, '.').replace(/[^\d.e-]+/gi, '')
```
[the remark this answers is not in the document]
Comment (btea) [reply packages/components/input-number/src/input-number.vue]: @E66Crisp As @coderabbitai said, we don't need to exclude e, since it has already been excluded.

[the remark this answers is not in the document]
Comment (btea) [reply packages/components/input-number/src/input-number.vue]: @coderabbitai Yeah, I misread it, your suggestion is great.

[the remark this answers is not in the document]
Comment (E66Crisp) [reply packages/components/input-number/src/input-number.vue]: nice suggestion

**Comment by rzzf** [comment]
To be honest, I’m a bit hesitant about changing `[type="number"]` to `[type="text"]`, as this would very likely introduce a breaking change.

The main reason is that switching to text brings in more edge cases and additional complexity. Based on what I’ve observed so far, here are a few examples of the differences:

1. When using scientific notation (e), number inputs only allow digits after `e`, while text inputs do not enforce this

2. After a decimal point is entered, number inputs prevent entering a second one, whereas text inputs allow it indefinitely

3. number inputs allow a leading `+`, while text inputs do not

…and other similar differences

On the other hand, switching to text does address some existing issues, such as https://github.com/element-plus/element-plus/pull/22382#discussion_r2401253418.

**Comment by E66Crisp** [comment]
> To be honest, I’m a bit hesitant about changing `[type="number"]` to `[type="text"]`, as this would very likely introduce a breaking change.
> 
> The main reason is that switching to text brings in more edge cases and additional complexity. Based on what I’ve observed so far, here are a few examples of the differences:
> 
> 1. When using scientific notation (e), number inputs only allow digits after `e`, while text inputs do not enforce this
> 2. After a decimal point is entered, number inputs prevent entering a second one, whereas text inputs allow it indefinitely
> 3. number inputs allow a leading `+`, while text inputs do not
> 
> …and other similar differences
> 
> On the other hand, switching to text does address some existing issues, such as [#22382 (comment)](https://github.com/element-plus/element-plus/pull/22382#discussion_r2401253418).

Thanks for the feedback.

The key point is: this change does solve the issue where users input only a decimal point (e.g., "1."), which seems unsolvable with `type="number"` due to browser's value sanitization behavior.

As for the other differences compared to type="number", I think they are acceptable as long as:

- They don't introduce other bugs
- They comply with our component's expected behavior

If there are specific edge cases you're concerned about, I'm happy to add test coverage or adjust the filtering logic accordingly.

**Comment by rzzf** [comment]
In fact, this behavior is caused by Chinese input methods, rather than an issue with input-number or `[type="number"]` itself.
When using a Chinese IME in any editable field, if you enter a number and then immediately type a period, the IME will usually convert it into a decimal point automatically. This explains why, after removing the period, it’s no longer possible to continue typing a period.

<hr />

**About `[type="text"]`:**

I looked into several mainstream Vue [component libraries](https://ui-libs.vercel.app/) and found that almost none of their numeric input components use `[type="number"]`. In addition, many articles express varying degrees of dissatisfaction with the default behavior of `[type="number"]`.

[Element UI](https://element.eleme.cn/#/zh-CN/component/input-number) also uses `[type="text"]`. After checking the Git history of Element Plus, I found that this change was introduced as part of a [feature request](https://github.com/element-plus/element-plus/issues/4017), primarily to trigger the numeric keyboard on mobile devices.
However, on iOS 18.1.1, this merely prioritizes a keyboard layout that includes numeric keys, and users can still switch to the alphabetic keyboard. Using only `inputmode="numeric"` or similar approaches does not seem to perfectly match the native behavior of `[type="number"]`. 

At the same time, some users expect input-number to accept only digits, decimal points, and `+/-` signs, and reject all other input (which is also why the `disabled-scientific` attribute was introduced).

- https://github.com/element-plus/element-plus/issues/8816
- https://github.com/element-plus/element-plus/issues/10629
- https://github.com/element-plus/element-plus/issues/16267
- https://github.com/element-plus/element-plus/issues/19514
- https://github.com/element-plus/element-plus/issues/20013
- https://github.com/element-plus/element-plus/issues/20522
- https://github.com/element-plus/element-plus/pull/21319

`[type="number"]` has been in use since version [1.2.0-beta.1](https://github.com/element-plus/element-plus/releases/tag/1.2.0-beta.1), which means it has been around for about four years.
For users who are already accustomed to this behavior, changing it would be a significant breaking change. That’s where my hesitation comes from. That said, I’m not opposed to this change—it would be ideal if we could keep the same constraints as number while using `[type="text"]`.
